src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -839,13 +842,14 @@ final class TrieMap[K, V] private (r: AnyRef, rtupd: AtomicReferenceFieldUpdater
   override def apply(k: K): V = {
     val hc = computeHash(k)
     val res = lookuphc(k, hc)
-    if (res eq null) throw new NoSuchElementException
+    if (res eq NoSuchElementSentinel) throw new NoSuchElementException
```

Comment:
I think `lookup` also needs to be adjusted. I don't know why that method is even public. And `getOrElseUpdate` seems to depend on `lookup` returning `null` for missing keys.